core/state/snapshot/generate.go

```diff
@@ -625,15 +628,15 @@ func (dl *diskLayer) generate(stats *generatorStats) {
 		if accMarker == nil || !bytes.Equal(accountHash[:], accMarker) {
 			dataLen := len(val) // Approximate size, saves us a round of RLP-encoding
 			if !write {
-				if bytes.Equal(acc.CodeHash, emptyCode[:]) {
+				if bytes.Equal(acc.PoseidonCodeHash, emptyPoseidonCode[:]) {
 					dataLen -= 32
```

Comment:
zktrie try to store every data field into 32 byte item, so it can parse each leaf and calculate the hash without extra knowledge for what it is (store value or account data or something else). This means we can not shrink a data field into 64bit unless we purpose another storage schema including the len of each item.

Reply:
ok, so in that case the `dataLen` should be reduced by 96?

Reply:
Omit that, you should be right